fighting/warrior.py

```diff
@@ -0,0 +1,138 @@
+import random
+from typing import Self
+
+
+class Armor:
+    def __init__(self, defence: int, class_armor: int):
+        self.class_armor = class_armor
+        self.armor: int | float = defence * self.multiplicity_check()
+
+    def multiplicity_check(self) -> float:
+        return {
+            1: 1,
+            2: 1.3,
+            3: 1.5,
+        }.get(self.class_armor, 1)
+
+    def _can_absorb_all_damage(self, damage: int | float) -> bool:
+        return damage <= self.armor
+
+    def absorb(self, damage: int) -> int | float:
+        if self._can_absorb_all_damage(damage):
+            self.armor -= damage
+            print(
+                f"The armor absorbed the damage successfully. Remaining armor is: {self.armor}"
+            )
+            return self.armor
+        else:
+            remaining_damage = self.armor - damage
+            self.armor = 0
+            print(
+                f"The armor don't absorb all the damage and part of the damage transfers to health. "
+                f"Remaining damage is: {abs(remaining_damage)}"
+            )
+            return remaining_damage
+
+
+class Weapon:
+    def __init__(
+        self,
+        name: str,
+        max_damage: int,
+        min_damage: int,
+        crit_chance: int,
+        crit_strange: int,
+    ):
+        self.name = name
+        self.min_damage = min_damage
+        self.max_damage = max_damage
+        self.crit_chance = crit_chance
+        self.crit_strange = crit_strange
+
+    def _is_crit_processed(self) -> int:
+        return random.randint(1, 100) <= self.crit_chance
+
+    def calculate_damage(self) -> int | float:
+        actual_damage = random.randint(self.min_damage, self.max_damage)
+        if self._is_crit_processed():
+            actual_damage *= (self.crit_strange / 100) + 1
+            print(f"Crit attack {actual_damage} damage")
+        else:
+            print(f"Non-critical damage is: {actual_damage} damage")
+        return actual_damage
+
+
+class Warrior:
+    def __init__(self, name: str, health: int, weapon: Weapon, armor: Armor):
+        self.name = name
+        self.health = health
+        self.weapon = weapon
+        self.armor = armor
+        self.is_alive: bool = True
+
+    def get_damage(self, damage: int):
```

Comment:
let's rename it to not be confused, as `get_damage` can be assumed as we calculate it and get, but not taking it
```suggestion
    def take_damage(self, damage: int):
```

Reply:
let's rename it to not be confused, as get_damage can be assumed as we calculate it and get, but not taking it
```suggestion
    def take_damage(self, damage: int):
```